apps/openssl.c

```diff
@@ -66,6 +66,7 @@ static int apps_startup(void)
 
     /* Set non-default library initialisation settings */
     if (!OPENSSL_init_ssl(OPENSSL_INIT_ENGINE_ALL_BUILTIN
+                          | CONF_MFLAGS_IGNORE_RETURN_CODES
```

Comment:
Huh? You are mixing flags here. This is not an `OPENSSL_INIT_*` flag. `CONF_MFLAGS_IGNORE_RETURN_CODES` has the value of 2, so this is equivalent to passing `OPENSSL_INIT_LOAD_CRYPTO_STRINGS` which also has the value of 2.

Reply:
I think what you mean to do is to create a new `OPENSSL_INIT_SETTINGS` object (via `OPENSSL_INIT_new()`), and call `OPENSSL_INIT_set_config_file_flags()` on it. The `OPENSSL_INIT_SETTINGS` object is then passed as the second argument to `OPENSSL_init_ssl` (current `NULL`).

Reply:
But since I'm looking for essentially default settings, wouldn't NULL do the same thing?

Reply:
I thought this was the reason for the `openssl list` failure in some of the CI runs, but perhaps is is something else...

Reply:
> But since I'm looking for essentially default settings, wouldn't NULL do the same thing?

Right. If you want default setting, then you don't need to do this at all, and just pass NULL.